keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -957,55 +961,84 @@ complete their service.
-exists while there is no actual RequestPriority object with `exempt ==
-true`.  The other is equivalent to the `workload-low` object exhibited
-above, and exists while there is no RequestPriority object with
-non-exempt priority.  There are also two implicitly generated
-FlowSchema backup objects.  Whenever a request whose groups include
-`system:masters` is not matched by any actual FlowSchema object, a
-backstop equivalent to the `system-top` object exhibited above is
-considered to exist.  Whenever a request whose groups do not include
-`system:masters` is not matched by any actual FlowSchema object, the
-following backstop object is considered to exist.
+There are three default behavior questions to answer: (1) what is the
+default configuration of a cluster, (2) how does an apiserver treat
+requests before it has had a chance to read the configuration and
+ensure it is reasonable, and (3) how is a request handled if it does
+not match any FlowSchema?
+
+#### Default Cluster Configuration
+
+There is a _predefined_ set of configuration objects, and it consists
+of the objects exhibited in the
+[Example Configuration](#example-configuration) section above.
+
+Whenever a kube-apiserver starts up, it checks whether there is a
+PriorityLevelConfiguration object with `Spec.Exempt==true`.  If not
+then the apiserver attempts to create the predefined configuration objects, the
+exempt PriorityLevelConfiguration last.  In this case: for each predefined
+object, the apiserver will create that object if there is not already
+an object with the same GroupKind and name; note that mere existence
+is enough; the apiserver does not insist that the spec matches its
+predefined content.
+
+Administrators can choose to persist a partially or completely
+different set of configuration API objects than the predefined set ---
+so long as the persisted set includes a PriorityLevelConfiguration
+object with `Spec.Exempt==true`.  (Remember that the behavior of the
+apiserver before it has had a chance to read the persisted
+configuration API objects is a different matter.)  Any configuration
+object can be created, updated, and deleted.
+
+#### API Server Initial Behavior
+
+As it starts up, an apiserver has to start handling requests before
+the filter discussed here has had a chance to read the persisted
+configuration API objects.  Indeed, that reading will have to be done
+through the very same filter (at least in the first apiserver in a
+cluster)!  The initial request handling also applies while an
+apiserver is persisting the default configuration API objects for the
+cluster.
+
+When the apiserver first starts up, it behaves as if the
+configuration consists of just (1) the predefined
+PriorityLevelConfiguration objects named `exempt` and `default`, and
+(2) the predefined FlowSchema objects named `exempt` and `default`.
+
+An apiserver maintains this initial behavior until it has had a
+chance to read all the persisted configuration API objects.
```

Comment:
Not just read (that might happen very fast when the collection is empty!) but when a valid set of configuration objects appears. Maybe it should continue this behavior until at least exempt and default objects appear.

Reply:
(I see this is at least partly covered below, maybe this is clear enough already)

Reply:
Yes, the next section is designed in such a way that we get that behavior without requiring any special code to extend the intial behavior period.